scalding-hadoop-test/src/test/scala/com/twitter/scalding/platform/PlatformTest.scala

```diff
@@ -342,15 +342,16 @@ class PlatformTest extends WordSpec with Matchers with HadoopSharedPlatformTest
       HadoopPlatformJobTest(new TypedPipeJoinWithDescriptionJob(_), cluster)
         .inspectCompletedFlow { flow =>
           val steps = flow.getFlowSteps.asScala
-          steps should have size 1
+          steps should have size 2
```

Comment:
Yeah so what's happening here as of now is that the rhs is a Pipe wrapping an IterableSource. Since we don't currently have a check for Pipe without parent, we end up falling back to checking if there's an Every before an Each and that ends up defaulting to false and we checkpoint (which is the extra step).

I can add the Pipe with no parent check in this location as well.